src/saml.ts

```diff
@@ -638,30 +668,18 @@ class SAML {
     );
   }
 
-  protected async certsToCheck(): Promise<string[]> {
-    let checkedCerts: string[];
-
+  protected async getKeyInfosAsPem(): Promise<string[]> {
     if (typeof this.options.cert === "function") {
-      checkedCerts = await util
-        .promisify(this.options.cert as CertCallback)()
-        .then((certs) => {
-          assertRequired(certs, "callback didn't return cert");
-          if (!Array.isArray(certs)) {
-            certs = [certs];
-          }
-          return certs;
-        });
-    } else if (Array.isArray(this.options.cert)) {
-      checkedCerts = this.options.cert;
-    } else {
-      checkedCerts = [this.options.cert];
+      // Do not cache
+      return await resolveAndParseKeyInfosToPem(this.options);
+    } else if (this.pemFiles.length > 0) {
+      // Return already cached PEM files.
+      return this.pemFiles;
```

Comment:
This is part of the new functionality that you've added that is completely uncovered by tests. Can you please add one here to make sure that caching is working as intended?